cedar-policy-core/src/entities/json/entities.rs

```diff
@@ -43,14 +46,22 @@ pub struct EntityJSON {
 
 /// Struct used to parse entities from JSON.
 #[derive(Debug, Clone)]
-pub struct EntityJsonParser<'e, S: Schema = NoEntitiesSchema> {
-    /// If a `schema` is present, this will inform the parsing: for instance, it
-    /// will allow `__entity` and `__extn` escapes to be implicit.
-    /// It will also ensure that the produced `Entities` fully conforms to the
-    /// `schema` -- for instance, it will error if attributes have the wrong
-    /// types (e.g., string instead of integer), or if required attributes are
-    /// missing or superfluous attributes are provided.
-    schema: Option<S>,
+pub struct EntityJsonParser<'e, 's, S: Schema = NoEntitiesSchema> {
+    /// `schema` represents a source of `Action` entities, which will be added
+    /// to the entities parsed from JSON.
+    /// (If any `Action` entities are present in the JSON, and a `schema` is
+    /// also provided, those entities' definitions must match exactly or an
+    /// error is returned.)
```

Comment:
```suggestion
    /// those entities' definitions must match exactly or an error is returned.)
```

Reply:
Updated the wording with a somewhat different change, hopefully things are clear now